### PR TITLE
stripchat: update m3u8 url

### LINKED ISSUE #7305: Stripchat: Error: No active streams found
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

UK

### Provide a description that is worded well enough to be understood

I've been trying all day to download streams but none are downloading.

C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2>TITLE WMDownload (1) 3

C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2>"C:\Users\User\AppData\Roaming\WM Recorder YT-DLP"\youtube-dl -o "C:\Users\User\Videos\WMR Recordings\Playlist Folders\WMR Recordings\13-%(title)s.%(ext)s"  --no-part --no-check-certificate -f best  --ignore-errors --ffmpeg-location "C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2\WM Converter Pro" "https://stripchat.com/Sunshine_Girll"
WARNING: "-f best" selects the best pre-merged format which is often not the best option.
         To let yt-dlp download and merge the best available formats, simply do not pass any format selection.
         If you know what you are doing and want only the best pre-merged format, use "-f b" instead to suppress this warning
[Stripchat] Extracting URL: https://stripchat.com/Sunshine_Girll
[Stripchat] Sunshine_Girll: Downloading webpage
ERROR: [Stripchat] Sunshine_Girll: Unable to find configuration for stream.; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2>pause
Press any key to continue . . .

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2>TITLE WMDownload (1) 3

C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2>"C:\Users\User\AppData\Roaming\WM Recorder YT-DLP"\youtube-dl -o "C:\Users\User\Videos\WMR Recordings\Playlist Folders\WMR Recordings\13-%(title)s.%(ext)s"  --no-part --no-check-certificate -f best  --ignore-errors --ffmpeg-location "C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2\WM Converter Pro" "https://stripchat.com/Sunshine_Girll"
WARNING: "-f best" selects the best pre-merged format which is often not the best option.
         To let yt-dlp download and merge the best available formats, simply do not pass any format selection.
         If you know what you are doing and want only the best pre-merged format, use "-f b" instead to suppress this warning
[Stripchat] Extracting URL: https://stripchat.com/Sunshine_Girll
[Stripchat] Sunshine_Girll: Downloading webpage
ERROR: [Stripchat] Sunshine_Girll: Unable to find configuration for stream.; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

C:\Program Files (x86)\WM Recorder-2\WM Recorder YT-2>pause
Press any key to continue . . .
```

### PR DESCRIPTION
**Pull request by foreignBlade**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fix the m3u8 url with the new format. For example

```
https://b-hls-23.doppiocdn.com/hls/92893973/92893973.m3u8?playlistType=lowLatency&_HLS_msn=3486&_HLS_part=0
```

ADD DESCRIPTION HERE

Fixes #7305

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🛠️🎥🆔

Fix HLS stream extraction for Stripchat by using model ID playlist instead of master playlist in `yt_dlp/extractor/stripchat.py`.

> _`HLS` stream URL_
> _changed to use `model ID` -_
> _fixes missing parts_

### Walkthrough
* Fix HLS stream URL to use model ID instead of master playlist ([link](https://github.com/yt-dlp/yt-dlp/pull/7306/files?diff=unified&w=0#diff-30808664de0951cafc7f7521d5ac7bdc79fc05c81554ad6bcf5e3f5f234cb44eL52-R52))

</details>

### CODE REVIEW

**Comment by bashonly** [comment]
you are reverting the url change from this PR: https://github.com/yt-dlp/yt-dlp/pull/5985/files#diff-30808664de0951cafc7f7521d5ac7bdc79fc05c81554ad6bcf5e3f5f234cb44eL47

it was changed because it is not the master manifest, it is a single variant (single resolution). can you find the master m3u8 url?

**Comment by mikeymatrix70** [comment]
Before I new there was an extractor to pull these I was manually getting the manifest URL.
for https://b-hls-23.doppiocdn.com/hls/92893973/92893973.m3u8?playlistType=lowLatency&_HLS_msn=3486&_HLS_part=0 the manifest url (master) would be 
https://b-hls-23.doppiocdn.com/hls/92893973/master/92893973_auto.m3u8
I always change the domain to edge-hls.doppiocdn.com and it always works, Had been doing this as xHamsterLive which uses the same feeds didn't have an extractor.  So to download that url I would use https://edge-hls.doppiocdn.com/hls/92893973/master/92893973_auto.m3u8 Hopefully someone can implement fix and also add xHamsterLive as an extractor

File: yt_dlp/extractor/stripchat.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def _real_extract(self, url):
         for host in traverse_obj(data, ('config', 'data', (
                 (('features', 'featuresV2'), 'hlsFallback', 'fallbackDomains', ...), 'hlsStreamHost'))):
             formats = self._extract_m3u8_formats(
-                f'https://b-{server}.{host}/hls/{model_id}/master/{model_id}_auto.m3u8',
+                f'https://b-{server}.{host}/hls/{model_id}/{model_id}.m3u8',
```
Comment (bashonly) [inline yt_dlp/extractor/stripchat.py:52]: remove the `server` var assignment above and:
```suggestion
                f'https://edge-hls.{host}/hls/{model_id}/master/{model_id}_auto.m3u8',
```

Comment (mikeymatrix70) [reply yt_dlp/extractor/stripchat.py:52]: That looks like that did the trick!

**Review by bashonly — APPROVED** [review]

**Comment by foreignBlade** [comment]
@mikeymatrix70 I realize that my credentials for spankbanglive work with stripchat, but not with xhamsterlive. But I would also like to see something that can download from "Collection".

**Comment by pro-ject143** [comment]
@bashonly Thanks for pointing out difference between yt-dlp and WM Recorder on Thursday. I understand now, and now using yt-dlp to download streams.

**Pull request merged**